### PR TITLE
Fix kubelet proxy configuration race condition

### PR DESCRIPTION
**Pull request by kke**
Trying to fix the race condition in `Configure kubelet proxy`

First round: alter the way initial node labels are set

### CODE REVIEW

**Comment by kke** [comment]
Specs and e2e pass, can't confirm if this fixes the 

```
ERROR: Pharos::PhaseManager::Error : Phase failed on 2 hosts:
- 667-do-charming-tapir:
RuntimeError: can't add a new key into hash during iteration
- 667-do-first-wildcat:
RuntimeError: can't add a new key into hash during iteration
```

File: lib/pharos/configuration/host.rb [hunk 1]
```diff
@@ -15,7 +15,7 @@ class Host < Pharos::Configuration::Struct
       attribute :private_address, Pharos::Types::Strict::String.optional.default(nil)
       attribute :private_interface, Pharos::Types::Strict::String.optional.default(nil)
       attribute :role, Pharos::Types::Strict::String
-      attribute :labels, Pharos::Types::Strict::Hash
+      attribute :labels, Pharos::Types::Strict::Hash.default(proc { {} })
```
Comment (jakolehm) [inline lib/pharos/configuration/host.rb]: Is this the minimal fix? Other changes are just extra?

Comment (kke) [reply lib/pharos/configuration/host.rb]: My understanding is that setting node labels in `def labels` is the problem. The changes to `initialize` make the host set the node role labels and taints already upon initialization.

**Comment by kke** [comment]
Looks like merge broke it.

**Pull request merged**